framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+
+      diagnosticsList.push_back(wasp::DataObject());
+
+      wasp::DataObject * diagnostic = diagnosticsList.back().to_object();
+
+      // according to the protocol line and column number are zero based
+
+      pass &= wasp::lsp::buildDiagnosticObject(*diagnostic,
+                                               this->errors,
+                                               line_number - 1,
+                                               column_number - 1,
+                                               line_number - 1,
+                                               column_number - 1,
+                                               1,
+                                               "moose_srv",
+                                               "check_inp",
+                                               error_line);
+    }
+  }
+
+  // reset behaviors of performance logging and error exception throwing
+
+  if (cached_logging_enabled)
+    Moose::perf_log.enable_logging();
+
+  Moose::_throw_on_error = cached_throw_on_error;
+
+  // remove temporary file that was dumped to the disk for parser access
+
+  std::remove(parse_file_path.c_str());
+
+  return pass;
+}
+
+bool
+MooseServer::updateDocumentTextChanges(const std::string & replacement_text,
+                                       int start_line,
+                                       int start_character,
+                                       int end_line,
+                                       int end_character,
+                                       int range_length)
+{
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // these are the expected values to indicate full document replacement
+
+  if (!(start_line == -1 && start_character == -1 && end_line == -1 && end_character == -1 &&
+        range_length == -1))
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs full text when changed" << std::endl;
+    return false;
+  }
+
+  this->document_text = replacement_text;
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentCompletionItems(wasp::DataArray & completionItems,
+                                           bool & is_incomplete,
+                                           int line,
+                                           int character)
+{
+  (void)completionItems; // suppress unused variable warning
+  (void)is_incomplete;   // suppress unused variable warning
+  (void)line;            // suppress unused variable warning
+  (void)character;       // suppress unused variable warning
```

Comment:
```suggestion
bool
MooseServer::gatherDocumentCompletionItems(wasp::DataArray & /*completionItems*/,
                                           bool & /*is_incomplete*/,
                                           int /*line*/,
                                           int /*character*/)
{
```

Reply:
Good feedback - I just added a commit addressing this.